TypeSystem/flow-analysis/promotion_via_assignment_A07_t01.dart

```diff
@@ -90,21 +90,24 @@ extension type ET2(int v) implements Object {}
 extension type ET3(int v) implements int {}
 
 test1(Object? o) {
-  if (o is ET1) {}
-  o = ET2(42);
-  o.expectStaticType<Exactly<Object>>();
+  if (o is ET1) {
+    o = ET2(42);
+    o.expectStaticType<Exactly<Object>>();
+  }
```

Comment:
```suggestion
  if (o is ET2) {} // Make `ET2` a type of interest.
  if (o is ET1) { // Promote `o` to `ET1`.
    o = ET2(42); // Everything except `ET2 <: ET1` is ok.
    o.expectStaticType<Exactly<Object>>(); // So we don't promote to `ET2`, only `Object`.
  }
```

Reply:
```dart
  o.expectStaticType<Exactly<Object>>(); // So we don't promote to `ET2`, only `Object`.
```
In fact, static type of `o` here is `ET2`. Is this a bug? Or may be we have missed anything?

Reply:
Thanks for going through with all this! ;-)